src/module.ts

```diff
@@ -0,0 +1,60 @@
+import { resolve } from 'path'
+import { fileURLToPath } from 'url'
+import { defu } from 'defu'
+import { defineNuxtModule, addPlugin } from '@nuxt/kit'
+
+export interface ModuleOptions {
+  /**
+   * Enable Device Module
+   * @default true
+   * @type boolean
+   */
+  enabled: boolean
+
+  /**
+   * Device Module Default User Agent
+   * @default 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_2) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/64.0.3282.39 Safari/537.36'
+   * @type string
+   */
+  defaultUserAgent: string,
+
+  /**
+   * Refresh Device Module Values On Window Resize
+   * @default false
+   * @type boolean
+   */
+  refreshOnResize: boolean
+}
+
+export default defineNuxtModule<ModuleOptions>({
+  meta: {
+    name: 'device-module',
+    configKey: 'device',
+    compatibility: {
+      nuxt: '^3.0.0-rc.11 || ^2.16.0',
+      bridge: true
+    }
+  },
+  defaults: {
+    enabled: true,
+    defaultUserAgent: 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_2) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/64.0.3282.39 Safari/537.36',
+    refreshOnResize: false
+  },
+  setup (options, nuxt) {
+    if (options.enabled) {
+      nuxt.options.runtimeConfig.public.device = defu(nuxt.options.runtimeConfig.public.device, {
```

Comment:
Not necessary to merge options manually ☺️ You can skip that block

Reply:
Thanks, removed it.

Reply:
More than understandable. Does/did the [module author guide](https://v3.nuxtjs.org/guide/going-further/modules) help you?

Reply:
It kinda did, maybe 5 months ago something was missing or wasn't explained clearly, I've checked it again and it seems fine.

Reply:
I was trying to implement the tests, and noticed that the removal of the merge options prevents the plugin to access the module options.

Inside `./src/runtime/plugin.ts` accessing `useRuntimeConfig().public` returns an empty object.

Reply:
Ohh, sorry. My bad! Didn't read correctly and thought you merge the options themself, not the `runtimeConfig`.